ulib/FStar.List.Tot.Base.fst

```diff
@@ -390,6 +390,18 @@ let rec unzip3 l = match l with
        let (tl1,tl2,tl3) = unzip3 tl in
        (hd1::tl1,hd2::tl2,hd3::tl3)
 
+(** Splitting a list at some index **)
+
+(** [first_N] takes a natural number n and a list and returns a pair
+    of the maximal prefix of l of size smaller than n and the rest of
+    the list *)
+let rec first_N (#a:Type) (n:nat) (l:list a) : list a * list a =
```

Comment:
I think Haskell calls this `splitAt`; I don't think Coq has it.  It could be neat to prove that it's the same as `(take n l, drop n l)`